pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -125,19 +136,44 @@ public void onBecomeOnlineFromConsuming(Message message, NotificationContext con
     public void onBecomeOfflineFromConsuming(Message message, NotificationContext context) {
       _logger.info("SegmentOnlineOfflineStateModel.onBecomeOfflineFromConsuming() : " + message);
       String realtimeTableName = message.getResourceName();
-      String segmentName = message.getPartitionName();
+      String segmentNameStr = message.getPartitionName();
+      TableDataManager tableDataManager = _instanceDataManager.getTableDataManager(realtimeTableName);
+      Preconditions.checkNotNull(tableDataManager);
+      if (tableDataManager instanceof RealtimeTableDataManager) {
```

Comment:
same. add a method in the base

Reply:
will like to not modify base class for now